functorch/csrc/DynamicLayer.cpp

```diff
@@ -513,6 +541,20 @@ void dynamicLayerFrontFallback(const c10::OperatorHandle& op, torch::jit::Stack*
 
   // Re-dispatch
   op.callBoxed(stack);
+  // checking exclude set because we should only do this if we actually hit a functionalization kernel.
+  if (layer.key() == DispatchKey::Functionalize && !exclude.has(DispatchKey::Functionalize)) {
+    auto ret_size = op.schema().returns().size();
+    foreachTensorInplace(*stack, stack->size() - ret_size, stack->size(),
+      [&](const Tensor& tensor) {
+        TORCH_INTERNAL_ASSERT(at::functionalization::impl::isFunctionalTensor(tensor));
+        auto wrapper = at::functionalization::impl::unsafeGetFunctionalWrapper(tensor);
+        // Functorch is responsible for setting the level on the wrapper, since we don't
+        // have that info available in core (for now).
+        wrapper->set_level(layer.layerId());
+        return tensor;
```

Comment:
Ugh this is pretty hacky, would it be better to have functional tensor in core propagate its level instead of rely on DispatchKey::Functionalize being in the exclude set?

Reply:
Totally agreed - I just can't think of a reasonable way this in core, at least for now. I'm hoping that we can handle this more gracefully if/when the dynamic layering logic makes its way into core.

The reason we can't really teach functional tensors to propagate their levels for now is really just.. factory operators. For non-factory ops, we can just read off the level from the input tensors to set the outputs. But when we hit a factory op, when you're in the functionalization kernel you have no way of knowing what level you are.